packages/picasso/src/RichTextEditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,237 @@
+      onBlur = noop,
+      placeholder,
+      minLength,
+      maxLength,
+      minLengthMessage,
+      maxLengthMessage,
+      style,
+      error,
+      testIds
+    },
+    ref
+  ) {
+    const classes = useStyles()
+    const toolbarRef = useRef<HTMLDivElement>(null)
+    const editorRef = useRef<HTMLDivElement>(null)
+    const wrapperRef = useRef<HTMLDivElement>(null)
+    const { dispatch, state } = useTextEditorState()
+
+    const { handleSelectionChange } = useOnSelectionChange({ dispatch })
+    const { handleTextFormat } = useOnTextFormat({ dispatch })
+    const {
+      handleBold,
+      handleItalic,
+      handleHeader,
+      handleOrdered,
+      handleUnordered
+    } = useToolbarHandlers({
+      editorRef,
+      handleTextFormat,
+      format: state.toolbar.format
+    })
+
+    const { isEditorFocused, handleFocus, handleBlur } = useOnFocus({
+      autoFocus,
+      editorRef,
+      toolbarRef,
+      wrapperRef,
+      onFocus,
+      onBlur,
+      dispatch
+    })
+
+    const defaultValueInHtml = useMemo(
+      // this effects needs to happen only once on first render
+      () =>
+        defaultValue
+          ? hastUtilToHtml(hastSanitize(defaultValue))
+          : defaultValue,
+      /* eslint-disable-next-line */
+      []
+    )
+
+    const { counterMessage, counterError, handleCounterMessage } = useCounter({
+      minLength,
+      maxLength,
+      minLengthMessage,
+      maxLengthMessage
+    })
+
+    return (
+      <Container
+        className={cx(
+          classes.editorWrapper,
+          {
+            [classes.disabled]: disabled,
+            [classes.focused]: isEditorFocused,
+            [classes.error]: error
+          },
+          className
+        )}
+        tabIndex={-1}
+        style={style}
+        ref={node => {
+          if (typeof ref === 'function') {
+            ref(node)
+          } else if (ref != null) {
+            ref.current = node
+          }
+          // @ts-ignore
+          wrapperRef.current = node
+        }}
+        data-testid={testIds?.wrapper || dataTestId}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+      >
+        <Toolbar
+          ref={toolbarRef}
+          disabled={disabled || state.toolbar.disabled}
+          id={id}
+          format={state.toolbar.format}
+          onBoldClick={handleBold}
+          onItalicClick={handleItalic}
+          onUnorderedClick={handleUnordered}
+          onOrderedClick={handleOrdered}
+          onHeaderChange={handleHeader}
+          testIds={{
+            headerSelect: testIds?.headerSelect,
+            boldButton: testIds?.boldButton,
+            italicButton: testIds?.italicButton,
+            unorderedListButton: testIds?.unorderedListButton,
+            orderedListButton: testIds?.orderedListButton
+          }}
+        />
+        <QuillEditor
+          ref={editorRef}
+          disabled={disabled!} // eslint-disable-line @typescript-eslint/no-non-null-assertion
```

Comment:
We could just cast this variable to boolean and remove the non-null-assertion

```suggestion
          disabled={!!disabled}
```

Reply:
✅